### ISSUE TITLE
piperack won’t use -Dcom.xmlcalabash.serial.indent=true

### ISSUE THREAD

**Issue by gimsieke** [ndw/xmlcalabash1#158]
Piperack 1.0.18 will respect p:serialization settings for indentation, but not said system property.
It might seem like a bad idea to specify an indentation setting for the whole server, but there is a use case…

**Comment by gimsieke** [comment]
Addition 1: If you have a primary and another output port, the 'indent' serialization property (maybe other serialization props, too) of the primary port will always be used for both ports, no matter what is specified for the non-primary port.
Repro: add `<p:serialization port="result2" indent="false"/>` to pipe.xpl as given on http://xmlcalabash.com/docs/reference/using-piperack.html 

Addition 2 (not exactly about indentation, but I don’t want to spoil your break with too many distinct bug reports, so I’m reusing this issue): if none of the ports in pipe.xsl is primary and if I post an input doc to the source port and then run the pipeline, it will tell me that there’s 1 document waiting on each port, that’s ok so far. But when I’m trying to GET any of them from `…/pipelines/…/outputs/result2` or from `…/pipelines/…/outputs/result`, I get the following error:

Exception or error caught in server resource

java.lang.NullPointerException
        at com.xmlcalabash.model.Step.getOutput(Unknown Source)
        at com.xmlcalabash.model.CompoundStep.getOutput(Unknown Source)
        at com.xmlcalabash.runtime.XPipeline.getSerialization(Unknown Source)
        at com.xmlcalabash.piperack.BaseResource.getOutput(Unknown Source)
        at com.xmlcalabash.piperack.Output.get(Unknown Source)
        ……

**Comment by ndw** [comment]
Sorry for the horrible delay. I'll try to get to these soon.

**Comment by ndw** [comment]
Hi Gerrit,

I can't reproduce the first issue. If I start piperack with a serialization option specified in a system property, it works for me. Can you reproduce that issue?

I just fixed the second issue, where the wrong port serialization parameters were being used.

I can't reproduce the third issue.

If you can reproduce either the first or third issue, please open a new issue for each of them. It's really much easier for me to track things if there's a separate issue for each defect. Thanks

/cc @gimsieke

**Comment by gimsieke** [comment]
I can’t reproduce the original issue, either.

**Comment by ndw** [comment]
I believe this is fixed in XML Calabash 1.0.19, released today.
